plugins/woocommerce-blocks/tests/e2e/tests/product-on-sale/product-on-sale.block_theme.spec.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * External dependencies
+ */
+import { expect, test } from '@woocommerce/e2e-playwright-utils';
+
+const blockData = {
+	name: 'On Sale Products',
+	slug: 'woocommerce/product-on-sale',
+};
+
+test.describe( `${ blockData.slug } Block`, () => {
+	test( 'can be inserted in Post Editor and it is visible on the frontend', async ( {
+		editorUtils,
+		editor,
+		admin,
+		frontendUtils,
+	} ) => {
+		await admin.createNewPost();
+		await editor.insertBlock( { name: blockData.slug } );
+		const blockLocator = await editorUtils.getBlockByName( blockData.slug );
+		await expect( blockLocator.getByRole( 'listitem' ) ).toHaveCount( 9 );
```

Comment:
This test seems to be failing. I think there are only 6 on-sale products instead of 9.

Reply:
Fixed with: [d1cee4f](https://github.com/woocommerce/woocommerce/pull/46671/commits/d1cee4f4e8c181a9de792dced8e5d72c40eb5fbc)